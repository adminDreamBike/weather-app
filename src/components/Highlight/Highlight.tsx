import { CardBody, Card, Text, Box, Progress, Icon } from "@chakra-ui/react";
import { FaLocationArrow } from "react-icons/fa6";

export const Highlight = ({
  title,
  value,
  units,
  progress,
  wind_deg,
}: {
  title: string;
  value: number;
  units: string;
  progress?: number;
  wind_deg?: number;
}) => {
  return (
    <Card background="#1e213a" w="52">
      <CardBody
        color="#e4e4e8"
        display="flex"
        flexDirection="column"
        alignItems="center"
      >
        <Text textAlign="center" fontSize="small">
          {title}
        </Text>
        <Box p="4" display="flex" flexDirection="row" py="2">
          <Text fontSize="4xl">{value}</Text>
          <Text fontSize="medium" alignContent="center">
            {units}
          </Text>
        </Box>
        {progress && <Progress value={progress} width="100px" size="sm" />}
        {wind_deg && (
          <Icon
            as={FaLocationArrow}
            style={{ transform: `rotate(${wind_deg}deg)` }}
          />
        )}
      </CardBody>
    </Card>
  );
};
