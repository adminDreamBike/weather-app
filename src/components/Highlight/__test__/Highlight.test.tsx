import { render, screen } from "@testing-library/react";
import { Highlight } from "../Highlight";
import { ChakraUIProvider } from "@/providers/ChakraUIProvider";

const defaultProps = {
  title: "Test",
  value: 130,
  units: "mph",
};

describe("", () => {
  test("should render Highlight component correctly", () => {
    const { asFragment } = render(
      <ChakraUIProvider>
        <Highlight {...defaultProps} />
      </ChakraUIProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Highlight with wind_deg and progres icon", () => {
    const { getByRole } = render(
      <ChakraUIProvider>
        <Highlight {...defaultProps} wind_deg={180} progress={80} />
      </ChakraUIProvider>
    );
    const progressIcon = getByRole("progressbar");

    expect(progressIcon).toBeDefined();
    expect(screen.getByRole("img")).toBeDefined();
  });
});
