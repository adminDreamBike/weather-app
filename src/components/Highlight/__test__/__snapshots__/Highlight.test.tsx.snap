// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Highlight component correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-card css-rrzei3"
  >
    <div
      class="chakra-card__body css-rzdfkz"
    >
      <p
        class="chakra-text css-15bto3y"
      >
        Test
      </p>
      <div
        class="css-skjq34"
      >
        <p
          class="chakra-text css-1bsg0uq"
        >
          130
        </p>
        <p
          class="chakra-text css-1v15ftq"
        >
          mph
        </p>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
